Generator/include/CGAL/point_generators_d.h

```diff
@@ -156,6 +156,9 @@ generate_point() {
       coord[i]=RT(this->d_range * ( 2 * this->_rnd.get_double() - 1.0));
 
     P p(dimension, coord.begin(), coord.end() );
+    #ifdef BOOST_GCC
+    #pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+    #endif
     this->d_item = p;
 }
```

Comment:
I agree with the patch, but this use of `#pragma GCC diagnostic ignored`. At the very least, you should use
```c++
#pragma GCC diagnostic push
/// ...
#pragma GCC diagnostic pop
```
To re-enable the warning once it is done.

Can you please point to the warning that these lines try to fix?

Reply:
oh! I thought the command only applies to the next instruction.
IIRC, the warning was in the `NewKernel_d` test.

Reply:
Ah no, this [one](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.14-Ic-61/Spatial_searching_Examples/TestReport_lrineau_ArchLinux-CXX17-Release.gz)
```
/usr/include/eigen3  -Wall -O3 -std=gnu++17 -frounding-math   -Wall -frounding-math -o CMakeFiles/fuzzy_range_query.dir/fuzzy_range_query.cpp.o -c /home/cgal_tester/build/src/cmake/platforms/ArchLinux-CXX17-Release/test/Spatial_searching_Examples/fuzzy_range_query.cpp
In file included from /home/cgal_tester/build/src/cmake/platforms/ArchLinux-CXX17-Release/test/Spatial_searching_Examples/fuzzy_range_query.cpp:2:
/mnt/testsuite/include/CGAL/point_generators_d.h: In member function 'void CGAL::Random_points_in_cube_d<P>::generate_point() [with P = CGAL::Wrap::Point_d<CGAL::Epick_d<CGAL::Dimension_tag<4> > >]':
/mnt/testsuite/include/CGAL/point_generators_d.h:159:5: warning: 'p' may be used uninitialized in this function [-Wmaybe-uninitialized]
     this->d_item = p;
     ^~~~
```

Reply:
@mglisse In the following code:
https://github.com/CGAL/cgal/blob/5b1e535cb96fed468e411e6429d18a5fa65cb7cd/Generator/include/CGAL/point_generators_d.h#L149-L160

with `P = CGAL::Wrap::Point_d<CGAL::Epick_d<CGAL::Dimension_tag<4> > >`, how can the compiler think that `p` at line 159 is not initialized? It is constructed the line before! Is there anything in `CGAL::Wrap::Point_d` that could explain that?